### PR TITLE
Fix submenu justification and spacer orientation.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

#36169 broke submenu alignment when the Navigation block is aligned right or space-between. This PR adds back the missing classnames as a temporary fix until we can sort out a better way to align submenus (#36241) 

It also patches the orientation context that Spacer block needs to work correctly inside Navigation. Again, this is a temporary measure until we find a better way to do this (#36197)

In doing this I noticed the Navigation block still has [an `is-vertical` classname](https://github.com/WordPress/gutenberg/blob/fix/submenu-alignment/packages/block-library/src/navigation/edit/index.js#L201) to which a few editor styles are added, mainly to do with the placeholder. I'm thinking these might no longer be needed since vertical/horizontal is now a setting instead of a block variation, so the placeholder will, in principle, never appear in vertical form. @jasmussen could we remove these styles?

## How has this been tested?

Add a Navigation block with some nested submenus (at least two levels of nesting). 
Align it right, and verify that 2nd level nested submenus are opening to the left instead of to the right.

Add a Spacer block inside the Navigation. 
Verify that in horizontal orientation, the Spacer grows horizontally.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Really nice, thank you. This fixes it in the editor right off the bat:
nav justifications

It doesn't appear to be working on the frontend:

Screenshot 2021-11-09 at 09 48 18

I'll dive in and see if I can push a fix for that.

**Comment by jasmussen** [comment]
> In doing this I noticed the Navigation block still has an is-vertical classname to which a few editor styles are added, mainly to do with the placeholder. I'm thinking these might no longer be needed since vertical/horizontal is now a setting instead of a block variation, so the placeholder will, in principle, never appear in vertical form. @jasmussen could we remove these styles?

About the placeholder, that's a good question, and goes to the heart of how we should design placeholders. The thing is, in a site editing context, the placeholder should ideally have a footprint that is _identical_ to that of the final result. Without it, [it will cause layout shifts](https://github.com/WordPress/gutenberg/issues/20853#issuecomment-621030013). 

There's wiggleroom of course — there are some blocks where the pre-configuraiton of key values is so crucial that it deserves up-front _weight_, but even then we have to consider the context in which they will be used, such as small viewports, or of the block is inserted inside a thin column. For example the navigation block in a column is already leveraging the built-in `is-small` classname that is being applied to the `Placeholder` by a resizeobserver:

Screenshot 2021-11-09 at 09 55 03

As far as block placeholders that look like the end result, my current favorite is the Site Logo:
site logo

Its footprint is identical, and it inherit colors and styles from the theme even in its unconfigured state. 

So back to your question, can we remove the `is-vertical` styles? **Yes, probably,** because most block setup states, _including navigation_, need to be revisited with a fresh perspective based on the challenges outlined above. In that vein, it's okay to pull back on the customization in the mean time. (Tangentially, #36302 is one tiny experiment in further reducing the footprint of the navigation setup state).

**Comment by jasmussen** [comment]
> It also patches the orientation context that Spacer block needs to work correctly inside Navigation. Again, this is a temporary measure until we find a better way to do this (#36197)

Thanks for working on this. I'm not sure what to look for, as trunk and this branch appear the same to me:
spacer

Some issues:

- Regardless of orientation, I can only resize the height. Likely the orientation related issue you've tracked in #36197.
- It's only 1px wide when freshly inserted, so it's hard to select. Upon clicking it, a width is set so it's easier to select.

It also made me realize that the mover controls are now all vertical, for all blocks inside the container, despite having horizontal orientation. This appears to be a general challenge for flex orientation blocks, as the Buttons block also shows vertical movers when it should be showing horizontal ones:

Screenshot 2021-11-09 at 10 05 47

Do you know if this is tracked separately? In any case, since these are general issues for the spacer and mover controls, they should not block this PR from fixing the justifications.

**Comment by jasmussen** [comment]
Back to the justifications which work in the editor but not in the frontend. It appears the classnames are working perfectly, the missing piece is that the `items-justified-n` classes are not output on the frontend. If I edit them in, things work:

inspectin

I suspect that's beyond my skill to fix. But it seems like once we add those classes, we can land this one. Thanks again!

**Comment by tellthemachines** [comment]
Ok I've fixed the justification classes on the front end, and the Spacer issue (there was no default value for orientation; I added one in so it should always work now). Also removed the `is-vertical` styles.

**Comment by jasmussen** [comment]
I'm having some local development environments that make me unable to test latest trunk, or this PR, at the moment. But if this works on the frontend, it'd be good to land, we can always follow up on things. It would be good with a quick sanity check by another.

**Review by jasmussen — APPROVED** [review]
Works well:
Screenshot 2021-11-11 at 08 37 14
Screenshot 2021-11-11 at 08 37 24

Thank you for the cleanup as well. As noted, I think I might be able to restore some of the vertical behavior for the placeholder when the orientation is changed — whether that's desirable remains to be seen, but no reason to lug the cruft along with it in the mean time.

**Pull request merged**
